Formula/lua@5.2.rb

```diff
@@ -0,0 +1,205 @@
+      url "https://luajit.org/patches/lua-5.2.0-advanced_readline.patch"
+      sha256 "33d32d11fce4f85b88ce8f9bd54e6a6cbea376dfee3dbf8cdda3640e056bc29d"
+    end
+  end
+
+  # sigaction provided by posix signalling power patch
+  if build.with? "sigaction"
+    patch do
+      # original patch file (not available): http://lua-users.org/files/wiki_insecure/power_patches/5.2/lua-5.2.3-sig_catch.patch
+      url "https://raw.githubusercontent.com/Homebrew/patches/d674e02d1097b21032198854080204680b616b61/lua/lua-5.2.3-sig_catch.patch"
+      sha256 "f2e77f73791c08169573658caa3c97ba8b574c870a0a165972ddfbddb948c164"
+    end
+  end
+
+  resource "luarocks" do
+    url "https://keplerproject.github.io/luarocks/releases/luarocks-2.3.0.tar.gz"
+    sha256 "68e38feeb66052e29ad1935a71b875194ed8b9c67c2223af5f4d4e3e2464ed97"
+  end
+
+  def install
+    # Subtitute formula prefix in `src/Makefile` for install name (dylib ID).
+    # Use our CC/CFLAGS to compile.
+    inreplace "src/Makefile" do |s|
+      s.remove_make_var! "CC"
+      s.change_make_var! "CFLAGS", "#{ENV.cflags} -DLUA_COMPAT_ALL $(SYSCFLAGS) $(MYCFLAGS)"
+      s.change_make_var! "MYLDFLAGS", ENV.ldflags
+    end
+
+    # Fix path in the config header
+    inreplace "src/luaconf.h", "/usr/local", HOMEBREW_PREFIX
+
+    # We ship our own pkg-config file as Lua no longer provide them upstream.
+    system "make", "macosx", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}", "INSTALL_INC=#{include}/lua-5.2"
+    system "make", "install", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}", "INSTALL_INC=#{include}/lua-5.2"
+    (lib/"pkgconfig/lua-5.2.pc").write pc_file
+
+    # Renaming from Lua to Lua52.
+    # Note that the naming must be both lua-version & lua.version.
+    # Software can't find the libraries without supporting both the hyphen or full stop.
+    mv "#{bin}/lua", "#{bin}/lua-5.2"
+    mv "#{bin}/luac", "#{bin}/luac-5.2"
+    mv "#{man1}/lua.1", "#{man1}/lua-5.2.4"
+    mv "#{man1}/luac.1", "#{man1}/luac-5.2.4"
+    bin.install_symlink "lua-5.2" => "lua5.2"
+    bin.install_symlink "luac-5.2" => "luac5.2"
+    include.install_symlink "lua-5.2" => "lua5.2"
+    (lib/"pkgconfig").install_symlink "lua-5.2.pc" => "lua5.2.pc"
+    (libexec/"lib/pkgconfig").install_symlink lib/"pkgconfig/lua-5.2.pc" => "lua.pc"
+
+    # This resource must be handled after the main install, since there's a lua dep.
+    # Keeping it in install rather than postinstall means we can bottle.
+    if build.with? "luarocks"
+      resource("luarocks").stage do
+        ENV.prepend_path "PATH", bin
+
+        system "./configure", "--prefix=#{libexec}", "--rocks-tree=#{HOMEBREW_PREFIX}",
+                              "--sysconfdir=#{etc}/luarocks52", "--with-lua=#{prefix}",
+                              "--lua-version=5.2", "--versioned-rocks-dir"
+        system "make", "build"
+        system "make", "install"
+
+        (pkgshare/"5.2/luarocks").install_symlink Dir["#{libexec}/share/lua/5.2/luarocks/*"]
```

Comment:
This one will need to however be changed back to `(share/"lua/5.2/luarocks")`

Reply:
does anything actually break?

Reply:
From memory, yes, but I can look into it properly at some point later probably.